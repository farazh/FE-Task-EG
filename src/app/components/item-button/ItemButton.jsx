import React from "react";
import "./ItemButton.scss";

export const ItemButton = ({ onItemAdd }) => {
  return (
    <button className="myButton" onClick={() => onItemAdd()}>
      ADD ITEM
    </button>
  );
};
