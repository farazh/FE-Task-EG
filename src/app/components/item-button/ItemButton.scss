.myButton {
  background-color: #91a3b0;
  border: 2px solid white;
  color: white;
  padding: 10px;
}
