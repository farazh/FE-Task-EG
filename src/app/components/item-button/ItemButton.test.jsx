import React from "react";
import { shallow } from "enzyme";
import { ItemButton } from "./ItemButton";

describe("ItemButton tests", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<ItemButton />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should fire onItemAdd when button is clicked", () => {
    const itemAdd = jest.fn();
    const wrapper = shallow(<ItemButton onItemAdd={itemAdd} />);
    wrapper.find("button").simulate("click");
    expect(itemAdd).toHaveBeenCalled();
  });
});
