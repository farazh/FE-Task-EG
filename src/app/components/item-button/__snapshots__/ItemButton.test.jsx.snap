// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemButton tests should match snapshot 1`] = `
<button
  className="myButton"
  onClick={[Function]}
>
  ADD ITEM
</button>
`;
